plugin/evm/vm_test.go

```diff
@@ -2110,3 +2113,116 @@ func TestCreateHandlers(t *testing.T) {
 		require.ErrorIs(t, elem.Error, rpc.ErrMissingBatchResponse)
 	}
 }
+
+// deployContract deploys the provided EVM bytecode using a prefunded test account
+// and returns the created contract address. It is reusable for any contract code.
+func deployContract(t *testing.T, vm *VM, gasPrice *big.Int, code []byte) common.Address {
+	callerAddr := vmtest.TestEthAddrs[0]
+	callerKey := vmtest.TestKeys[0]
+
+	nonce := vm.txPool.Nonce(callerAddr)
+	tx := types.NewContractCreation(nonce, big.NewInt(0), 1000000, gasPrice, code)
+	signedTx, err := types.SignTx(tx, types.NewEIP155Signer(vm.chainConfig.ChainID), callerKey.ToECDSA())
+	require.NoError(t, err)
+
+	for _, err := range vm.txPool.AddRemotesSync([]*types.Transaction{signedTx}) {
+		require.NoError(t, err)
+	}
+
+	blk, err := vm.BuildBlock(context.Background())
+	require.NoError(t, err)
+	require.NoError(t, blk.Verify(context.Background()))
+	require.NoError(t, vm.SetPreference(context.Background(), blk.ID()))
+	require.NoError(t, blk.Accept(context.Background()))
+
+	ethBlock := blk.(*chain.BlockWrapper).Block.(*wrappedBlock).ethBlock
+	receipts := vm.blockChain.GetReceiptsByHash(ethBlock.Hash())
+	require.Len(t, receipts, len(ethBlock.Transactions()))
+
+	found := false
+	for i, btx := range ethBlock.Transactions() {
+		if btx.Hash() == signedTx.Hash() {
+			found = true
+			require.Equal(t, uint64(1), receipts[i].Status) // 1 = success
+			break
+		}
+	}
+	require.True(t, found)
+
+	return crypto.CreateAddress(callerAddr, nonce)
+}
+
+func TestMakePrecompileRevert_Granite(t *testing.T) {
```

Comment:
Can we merge this test with Fortuna and have test-table like cases for:
1- granite = should revert 
2- pre-granite (fortuna) = should invalidate and return err
3- pre-granite + pre-invalidateDelegateUnix = should do be sucessfull